src/pages/non-pass-finalisation/cat-d/non-pass-finalisation.cat-d.page.ts

```diff
@@ -0,0 +1,235 @@
+import { Component, OnInit } from '@angular/core';
+import { IonicPage, NavController, Platform } from 'ionic-angular';
+import { AuthenticationProvider } from '../../../providers/authentication/authentication';
+import { Store, select } from '@ngrx/store';
+import { StoreModel } from '../../../shared/models/store.model';
+import { CAT_D } from '../../page-names.constants';
+import { Observable } from 'rxjs/Observable';
+import { getTests } from '../../../modules/tests/tests.reducer';
+import {
+  getCurrentTest,
+  getJournalData,
+  getActivityCode,
+  getTestOutcome,
+  isTestOutcomeSet,
+  getTestOutcomeText,
+} from '../../../modules/tests/tests.selector';
+import { getCandidate } from '../../../modules/tests/journal-data/cat-d/candidate/candidate.cat-d.reducer';
+import {
+  getUntitledCandidateName,
+  getCandidateDriverNumber,
+  formatDriverNumber,
+} from '../../../modules/tests/journal-data/common/candidate/candidate.selector';
+import {
+  NonPassFinalisationViewDidEnter,
+  NonPassFinalisationValidationError,
+} from '../non-pass-finalisation.actions';
+import { map, withLatestFrom, tap } from 'rxjs/operators';
+import { getTestSummary } from '../../../modules/tests/test-summary/test-summary.reducer';
+import { isDebriefWitnessed, getD255 } from '../../../modules/tests/test-summary/test-summary.selector';
+import {
+  getTestSlotAttributes,
+} from '../../../modules/tests/journal-data/common/test-slot-attributes/test-slot-attributes.reducer';
+import { isWelshTest }
+  from '../../../modules/tests/journal-data/common/test-slot-attributes/test-slot-attributes.selector';
+import {
+  ActivityCodeModel,
+  activityCodeModelList,
+} from '../../office/components/activity-code/activity-code.constants';
+import { FormGroup } from '@angular/forms';
+import { PersistTests } from '../../../modules/tests/tests.actions';
+import { OutcomeBehaviourMapProvider } from '../../../providers/outcome-behaviour-map/outcome-behaviour-map';
+// Todo: change to cat d when it is availble
+import { behaviourMap } from '../../office/office-behaviour-map.cat-c';
```

Comment:
If Ross merges his PR first you could catch this in a rebase (or the other way around)

Reply:
i have checked with Ross and cat-d is now available. so i have made the change